presto-spi/src/main/java/com/facebook/presto/spi/block/UncheckedBlock.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.api.Experimental;
+import io.airlift.slice.Slice;
+
+/**
+ * Accessors for Block which provide for raw indexing into the underlying data structure of the Block.
+ * It is intended for hot path use cases where the speed of the boundary checks on Block are a performance
+ * bottleneck.  Methods in this case must be used with more care, as they may return invalid data if the
+ * indexing is not performed with respect to {@link #getOffsetBase} and {@link #getPositionCount()}.
```

Comment:
It might be helpful to provide an example. E.g. say getOffsetBase() returns 5 and getPositionCount() returns 100, then there are 100 valid entries and they should be accessed using positions from [5, 105) range, e.g.

```
for (int i = offsetBase; i < offsetBase + positionCount; i++) {
    block.getByte(i);
}
```

Did I get it right?